### ISSUE TITLE
Subject-Location Restriction

### ISSUE THREAD

**Issue by sandhawke** [solid/solid#81]
Some people understand Linked Data to be suggesting that if the RDF triple { ?S ?P ?O } exists, it can be found my referencing ?S.

More formally, we could say:   If an RDF Source at URL ?G adheres to the Subject-Location Restriction, and it provides an RDF graph containing a triple { ?S ?P ?O }, then truncateAtHash(?S) == ?G.

Now, how should we treat this restriction?

Is this:

1. A pattern that apps MAY use if they want
2. A best-practice that apps SHOULD follow, but pods MUST NOT enforce
3. A rule that all apps MUST follow and pods MAY enforce
4. A rule that all apps MUST follow and pods MUST enforce

Note that currently gold and ldnode do not enforce this, but meccano.io does.

**Comment by sandhawke** [comment]
One problem, described by @deiu, is a person having a public foaf:name for themselves and private contact information for themselves.   Same ?S, different ACLs, so different ?G.   

A workaround is to use inference.    Have multiple IRIs for the person,  and say they are owl:sameAs each other.

**Comment by sandhawke** [comment]
Advantages:
* you can just use a triple store, instead of a quad store (no need for Named Graphs)
* a model more like nested data structures

**Comment by sandhawke** [comment]
Problem:

* Archiving or Caching.   Making a copy of someone else's RDF on your server

**Comment by sandhawke** [comment]
Problem:
* Can't store intermediate, imperfect, being-edited data

**Comment by melvincarvalho** [comment]
> Some people understand Linked Data to be suggesting that if the RDF triple { ?S ?P ?O } exists, it can be found by dereferencing ?S.

I wasnt aware of this debate.  Is it considered a best practice?

As a fan of decentralization I like the mantra of on the web 'anyone can say anything about anything'.  Is this not a mainstream approach -- I dont really know the full history of this issue.

What I remember @bblfish once saying is that the HTTP URL of a subject is called 'definitive'.  

On a practical note, one thing I sometimes do is pull out my facebook contacts from facebook, and store them in my knowledge base (e.g. lets say prefs file for the sake of argument).  I can then augment that data with useful bits and pieces such as name avatar and app preferences, in order to onboard my facebook friends, invite them to use solid apps, to chat etc. (for example with a capability URL passed out of band) so this is done in a way that need not go through any central points of control, meaning users really do own their own data in a social way, rather than servers owning it.

**Comment by akuckartz** [comment]
This is closely related to "Follow your nose" (https://www.w3.org/2001/sw/wiki/Linking_patterns#.E2.80.9CFollow_your_nose.E2.80.9D).

**Comment by sandhawke** [comment]
Yes, it's taking taking Follow Your Nose a bit more seriously than it was originally meant.   Instead of FYN being a suggestion, it turns into that's all you have to do.   Which would be nice, if it turns out to work.

**Comment by sandhawke** [comment]
Possible middle ground: one could have asserted-RDF and unasserted-RDF on one's server.  Unasserted-RDF is just bytes, really.  It can have typos.

**Comment by dmitrizagidulin** [comment]
I'm curious how the whole 'Follow Your Nose' terminology got started :) Like, why isn't it simply referred to as link following, etc.

**Comment by deiu** [comment]
It's a way of saying "oh hey, this looks interesting, I wonder it will lead me to".

**Comment by dmitrizagidulin** [comment]
> It's a way of saying "oh hey, this looks interesting, I wonder it will lead me to".

Does that mean you started it? :)

**Comment by sandhawke** [comment]
Dan Connolly started it, I think, circa 2001.    He used that term because "link following" was more ambiguous.   I mean, *why* do you follow a link?     I've always assumed it was a reference, in part, to the Fruit Loops commercials, although it makes sense even without that.

**Comment by dckc** [comment]
The Fruit Loops thing was subliminal at most.

To me, "follow your nose" means: look right in front of you. Take the clearly evident step forward.

In the web context, it means: when looking at `<link href="http://domain/path" />` in an HTML document (or other syntax defined to carry URLs), there are, broadly, two options:
  1. look up `http://domain/path` in some context not evident in what you're looking at
     - for example, do a google search for `link:http://domain/path`
  2. look the name up based on how URLs are defined to work; i.e. look up `domain` in DNS, connect to port 80, and `GET /path`

Following your nose is the latter.

**Comment by bblfish** [comment]
> Some people understand Linked Data to be suggesting that if the RDF triple { ?S ?P ?O } exists, it can be found by dereferencing ?S.

I don't know anyone who suggests that.  That would mean that one document would contain all the triples about a specific subject, which is clearly false.

What makes more sense is that if you want to know the meaning of a URL you dereference it. How this works is given in [the WebID spec](https://www.w3.org/2005/Incubator/webid/spec/identity/):

 ![](https://www.w3.org/2005/Incubator/webid/spec/identity/img/WebID-overview.png)
